common/src/main/java/io/netty/util/concurrent/GlobalEventExecutor.java

```diff
@@ -104,15 +104,14 @@ Runnable takeTask() {
     }
 
     private void fetchFromScheduledTaskQueue() {
-        if (hasScheduledTasks()) {
-            long nanoTime = AbstractScheduledEventExecutor.nanoTime();
-            for (;;) {
-                Runnable scheduledTask = pollScheduledTask(nanoTime);
-                if (scheduledTask == null) {
-                    break;
-                }
-                taskQueue.add(scheduledTask);
+        long nanoTime = AbstractScheduledEventExecutor.nanoTime();
+        Runnable scheduledTask = pollScheduledTask(nanoTime);
+        while (scheduledTask != null) {
+            if (scheduledTask == null) {
```

Comment:
You're right that's not necessary. I missed removing that when I refactored from "for" to "while". I'll adjust the PR.